batched/dense/unit_test/Test_Utils.hpp

```diff
@@ -0,0 +1,105 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+#ifndef TEST_UTILS_HPP
+#define TEST_UTILS_HPP
+
+#include <Kokkos_Core.hpp>
+#include <iomanip>
+#include "KokkosBatched_Util.hpp"
+
+template <typename ExecutionSpace, typename AViewType, typename BViewType>
+bool allclose(const AViewType& a, const BViewType& b, double rtol = 1.e-5,
```

Comment:
Did you check in the `test_common` and `common` directories for these utilities? If we already have them there we don't want to re-implement this

Reply:
I have made a look at test_common and common, but cannot find the same functionalities. 
1. `allclose` is useful to make an element wise error check over multi dimensional views on device
2. `create_banded_triangular_matrix` is more important since we need a helper to create a banded matrix for testing

Further comments or suggestions are highly appreciated

Reply:
How about this: `EXPECT_NEAR_KK_1DVIEW` or this `EXPECT_NEAR_KK_REL_1DVIEW` for `KokkosKernels_TestUtils.hpp`? It looks similar to me, maybe you can just expand these a bit to add what you need? It's just usually better not to rewrite them every time to reduce bloat and potential errors.

Regarding `create_banded_triangular_matrix `, I agree this is new and needed so no issue there.

Reply:
Sure. Maybe I will ask for something like `EXPECT_NEAR_KK_NDVIEW` and `EXPECT_NEAR_KK_REL_NDVIEW`, in another PR. For the moment, I will remove `allclose` and do the error check on the host. 

As for `create_banded_triangular_matrix`, I will move this to `Test_Batched_DenseUtils.hpp` because similar function is there.